packages/lit-html/src/lit-html.ts

```diff
@@ -350,63 +367,96 @@ class Template {
         regex.lastIndex = lastIndex;
         match = regex.exec(s);
         if (match === null) {
-          // TODO (justinfagnani): add test coverage from spread parts
-          // If we're not in a quoted attribute value, make sure we clear
-          // the attrNameEnd marker
-          if (regex !== singleQuoteAttr && regex !== doubleQuoteAttr) {
-            attrNameEnd = -1;
+          // If the current regex doesn't match we've come to a binding inside
+          // that state and must break and insert a marker
+          if (regex === tagEndRegex) {
```

Comment:
This is a better check. The only time we could have a positive `attrNameEndIndex` is when regex was `tagEndRegex`, `doubleQuoteAttrEndRegex`, or `singleQuoteAttrEndRegex`. The assertion after the while loop checks that.

Reply:
ok, makes sense.